Test

Asdfif opened the pull request.

Asdfif committed:
Созданы основные классы игры

класс Card  - имеет масти, значения и названия (значение + масть)

класс Deck - инстанс класса содержит массив всех возможных карт

класс Hand - имеет массив, в который помещаются карты. При взятии карты в руку из колоды, карта помещается в руку и удаляется из колоды

класс Player и подкласс Dealer - имеет имя, id, игровой статус(проиграл, если сумма очков карт в руке больше 21), стартовую сумму денег. Имеют методы bet - сделать ставку, take_hand - два раза добавить карту в руку, take_card - взять добавить карту в руку,  skip - ничего не делать, calculate - посчитать сумму очков карт в руке. Дилер имеет метод decision - принятие решение какой метод использовать в зависимости от текущих очков.

Asdfif committed:
Изменен метод calculate

метод calculate перенесен из класса Player в класс Hand

Asdfif committed:
изменен метод calculate

в методе уменьшено ветвление

Asdfif committed:
Revert "изменен метод calculate"

This reverts commit 2b7b336caef1fb725d98982e5a621ddc7f5c0b82.

Asdfif committed:
изменен метод calculate

уменьшено ветвление в методе calculate

Asdfif committed:
Добавлены классы Game и Interface

класс Game - механика игры

класс Interface - интерфейс игры, вывод информации на экран

russo-matrosso commented on card.rb:
```diff
@@ -0,0 +1,16 @@
+  def initialize(nominal, suit)
+    @card_nominal = nominal
+    @card_suit = suit
+    @card_name = "#{@card_nominal}#{@card_suit} "
```
В строку должен преобразовывать интерфейс

russo-matrosso commented on dealer.rb:
```diff
@@ -0,0 +1,18 @@
+class Dealer < Player
+  def initialize
+    super('Dealer')
+    @id = 0
```
А зачем нам id?

russo-matrosso commented on deck.rb:
```diff
@@ -0,0 +1,19 @@
+
+  def initialize
+    @all_cards = []
+    suits = Card.const_get('SUITS')
```
Не надо использовать метапрограммирование.
Card::SUITS - тоже самое

russo-matrosso commented on deck.rb:
```diff
@@ -0,0 +1,19 @@
+    @all_cards = []
+    suits = Card.const_get('SUITS')
+    values = Card.const_get('VALUES')
+    (0..(values.size - 1)).each do |i|
```
`Card::VALUES.each do` итд

russo-matrosso commented on game.rb:
```diff
@@ -0,0 +1,134 @@
+attr_accessor :players, :bank, :queue, :deck, :table, :score, :winners, :interface
+
+	def initialize(name, interface)
+		@interface = interface
```
Табы в качестве отступов - не ОК

russo-matrosso commented on game.rb:
```diff
@@ -0,0 +1,134 @@
+		@interface = interface
+		@players =[]
+		@queue=[]
+		@players = Array.new
```
Используй рубокоп. Array.new - не ОК

russo-matrosso commented on hand.rb:
```diff
@@ -0,0 +1,39 @@
+
+  def initialize(deck)
+    @cards = []
+    add_card(deck)
```
2.times do

Asdfif committed:
Внесены изменения по замечаниям

название карты теперь выводится через интерфейс, удален атрибут card_name
удален атрибут id, потому что он не особо важен
применен рубокоп

russo-matrosso commented on deck.rb:
```diff
@@ -7,13 +7,10 @@ class Deck
-    (0..(values.size - 1)).each do |i|
-      (0..(suits.size - 1)).each do |j|
-        @all_cards << Card.new(values[i], suits[j])
+    (0..(Card::VALUES.size - 1)).each do |i|
```
Здесь точно не стоит использовать range - `Card::VALUES.each do |card|` работает также, но без индекса

russo-matrosso commented on game.rb:
```diff
@@ -1,134 +1,130 @@
+    @interface = interface
+    @players = []
+    @queue = []
+    @players = []
```
А зачем две одниаковыйе переменные players?

Asdfif committed:
изменен Deck

убран range при создании колоды
также убран лишний @players=[]
